Std/Tactic/PrintPrefix.lean

```diff
@@ -4,36 +4,147 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Shing Tak Lam, Daniel Selsam, Mario Carneiro
 -/
 import Std.Lean.Util.EnvSearch
-import Lean.Elab.Command
+import Lean.Elab.Tactic.Config
 
 namespace Lean.Elab.Command
 
-private def appendMatchingConstants (msg : String)
-    (ϕ : ConstantInfo → MetaM Bool) (opts : EnvironmentSearchOptions := {}) : MetaM String := do
-  let cinfos ← getMatchingConstants ϕ opts
-  let cinfos := cinfos.qsort fun p q => p.name.lt q.name
-  let mut msg := msg
-  for cinfo in cinfos do
-    msg := msg ++ s!"{cinfo.name} : {← Meta.ppExpr cinfo.type}\n"
-  pure msg
+/--
+Options to control `#print prefix` command and `getMatchingConstants`.
+-/
+structure PrintPrefixConfig where
+  /-- Include declarations in imported environment. -/
+  imported : Bool := true
+  /-- Include declarations whose types are propositions. -/
+  propositions : Bool := true
+  /-- Exclude declarations whose types are not propositions. -/
+  propositionsOnly : Bool := false
+  /-- Print the type of a declaration. -/
+  showTypes : Bool := true
+  /--
+  Include internal declarations (names starting with `_`, `match_` or `proof_`)
+  -/
+  internals : Bool := false
+
+/-- Function elaborating `Config`. -/
+declare_config_elab elabPrintPrefixConfig PrintPrefixConfig
 
 /--
 Command for #print prefix
 -/
-syntax (name := printPrefix) "#print prefix " ident : command
+syntax (name := printPrefix) "#print prefix " (Lean.Parser.Tactic.config)? ident : command
```

Comment:
Do we really want to require exactly one space between `#print` and `prefix`? That seems excessively rigid.

Reply:
Good catch.  I fixed this.  I'm also not a huge fan of the config syntax I used from `Tactic`, but some small changes I tried didn't make it better, so I am keeping as is.

Reply:
FWIW, I think the fix allows things like:
```lean
#print 

prefix 

Empty
```
I don't think this is a major problem though, no fix necessary.